AppController/djinn.rb

```diff
@@ -4774,6 +4774,9 @@ def prep_app_dashboard()
       setup_app_dir(AppDashboard::APP_NAME, true)
     }
 
+    # Allow fewer dashboard instances for small deployments.
+    min_dashboards = [3, get_all_appengine_nodes.length].min
```

Comment:
We could, but the assignment here is permanent. Often, we have plenty of memory shortly after starting AppScale, but the available memory is reduced after running the first hawkeye test. The number of appengine machines can also change after this assignment (so neither assignment method is great), but that number is less volatile than the number of AppServers we can run at the moment.